test/examples/test_bart.py

```diff
@@ -0,0 +1,248 @@
+import math
+
+import torch
+
+import funsor.distributions as dist
+import funsor.ops as ops
+from funsor.cnf import Contraction
+from funsor.domains import bint, reals
+from funsor.gaussian import Gaussian
+from funsor.integrate import Integrate
+from funsor.interpreter import interpretation
+from funsor.montecarlo import monte_carlo
+from funsor.pyro.convert import AffineNormal
+from funsor.terms import Binary, Independent, Number, Reduce, Slice, Stack, Subs, Variable, reflect
+from funsor.torch import Function, Tensor
+
+num_origins = 2
+num_destins = 2
+
+
+def bounded_exp(x, bound):
+    return (x - math.log(bound)).sigmoid() * bound
+
+
+def unpack_gate_rate(gate_rate):
+    batch_shape = gate_rate.shape[:-1]
+    event_shape = (2, num_origins, num_destins)
+    gate, rate = gate_rate.reshape(batch_shape + event_shape).unbind(-3)
+    rate = bounded_exp(rate, bound=1e4)
+    gate = torch.stack((torch.zeros_like(gate), gate), dim=-1)
+    return gate, rate
+
+
+def unpack_gate_rate_0(gate_rate):
+    return unpack_gate_rate(gate_rate)[0]
+
+
+def unpack_gate_rate_1(gate_rate):
+    return unpack_gate_rate(gate_rate)[1]
+
+
+def test_bart():
```

Comment:
I expect tests like these to be short lived, but they are invaluable as @eb8680 and @jpchen and @martinjankowiak and I collaborate on experiments.